src/d3a/d3a_core/util.py

```diff
@@ -486,3 +486,30 @@ def is_time_slot_in_past_markets(time_slot: DateTime, current_time_slot: DateTim
             hours=ConstSettings.SettlementMarketSettings.MAX_AGE_SETTLEMENT_MARKET_HOURS))
     else:
         return time_slot < current_time_slot
+
+
+class FutureMarketCounter:
+    """Hold a time counter for the future market.
+
+    In the future market, we only want to clear in a predefined interval.
+    """
+    def __init__(self):
+        self.last_time_dispatched = None
+
+    def __call__(self, last_time_dispatched: DateTime):
+        """To be called when activating the counter."""
+        self.last_time_dispatched = last_time_dispatched
+
+    def is_time_for_clearing(self, current_time: DateTime) -> bool:
+        """Compare current time with the latest time clearing was dispatched.
+
+        Returns True if the FUTURE_MARKET_CLEARING_INTERVAL_MINUTES has
+        already passed since the last dispatch time.
+        """
```

Comment:
```suggestion
        """
        if not self.last_time_dispatched: 
            return True
```
In order to immediately mark that it is time for clearing if the original value is None.

Also minor but it would be good to be added, last_time_dispatched can probably be private.

Reply:
@spyrostz seems great, do you think that we can also get rid of the `__call__` method and set the `last_time_dispatched` in the first function call as following  
```python
if not self._last_time_dispatched: 
            self._last_time_dispatched = current_time
            return True

```

Reply:
I suppose the only difference we will get is at the first call, do we want to match future markets from the very beginning or do we need to wait the interval before that ?

Reply:
@Hassan754 even better, I like your suggestion even more!